### ISSUE TITLE
Use a Database System for quotes.

### ISSUE THREAD

**Issue by AnonGuy** [CyberDiscovery/cyberdisc-bot#50]
The current implementation of quote queries, #41, raises the issue of quote storage. The implementation requires the bot to iterate every message in the specified quotes channel, storing these results temporarily and using `random.choice` to select a quote. The implementation wasn't originally @picapi's idea, I'm just using it as an example.

Clearly, the feature works, but it is quite slow. The performance loss will only get more apparent as more quotes are added, and it's quite an unreliable method as is. I recommend implementing a Database for the bot, where quotes are persistently stored in a table, e.g. `quotes`, and a random record can be selected with a simple SQL query:
```sql
SELECT content FROM quotes 
ORDER BY RANDOM() LIMIT 1
```

The easiest DBMS to use to implement this is `sqlite`, and for an async application like this bot, we'll need to use the `aiosqlite` library for asynchronous queries. This will make the quotes command **much** faster and more responsive. It also enables us to interface with the database via other cogs.

I'd be happy to implement this in a PR, but I'd like to hear some thoughts from contributors beforehand.

**Comment by fractalpenguin** [comment]
If you want to implement it, go ahead: I'll be glad to approve such a change.

**Comment by thebeanogamer** [comment]
I'm in the same boat as @fractalpenguin. I've got no idea how I would go about creating somthing like this, but if you do and are willing to put in the time and effort, it would be very much appreciated.

**Comment by AnonGuy** [comment]
Okay, great. It might take a while for me to get round to this, but I'll make a PR when I do.

**Comment by NateKomodo** [comment]
Tried it out today and I am in agreement.

**Comment by ed588** [comment]
Would you consider using something like redis rather than sqlite?

**Comment by AnonGuy** [comment]
@ed588 Mainly because I don't have much experience with NoSQL, and having some relational database features might be useful. I'm still willing to implement this, though, if it's beneficial. Are there any reasons you'd prefer to use Redis over Sqlite?

**Comment by ed588** [comment]
No, I was just wondering. I could implement it using sqlite if you want.

**Comment by AnonGuy** [comment]
Ah, I see. If you're interested in contributing, feel free to create a PR for a `good first issue`, and get it merged. That's a quick way to get Community Dev, and you'll be able to push to the `database-system` branch in no time.

**Comment by TotallyCoded** [comment]
Is anyone actively working on this issue? If not I might give it a go this weekend.

**Comment by AnonGuy** [comment]
After some discussion with @aquariuscd, I think it's safe to say that we don't really need to implement a database system for this feature: we only really need to be able to store `{user_id: [message_id, ...], ...}`, and our current 1,500 quotes don't really warrant a database. We could just create this dictionary as the bot loads, and sample the IDs for the quote commands.

**Comment by AnonGuy** [comment]
@aquariuscd Will be implementing this in a PR.
